packages/core-app-api/src/apis/implementations/IdentityApi/AppIdentityProxy.ts

```diff
@@ -130,10 +130,9 @@ export class AppIdentityProxy implements IdentityApi {
 
   async signOut(): Promise<void> {
     await this.waitForTarget.then(target => target.signOut());
-
     await this.#cookieAuthSignOut?.();
 
-    window.location.href = this.signOutTargetUrl;
+    window.location.href = `${this.signOutTargetUrl}?signedOut=true`;
```

Comment:
Thinking it might be better to include the query param when we set the initial `signOutTargetUrl`, or we need to parse it here and add it that way instead. The `signOutTargetUrl` might already contain query parameters, which would break this.

I do wonder if it's working using `sessionStorage` for this, setting a flag that is picked up by the sign-in page and using a timestamp to make sure not too long ago. What do you think about that approach?

Reply:
Instead of appending query parameters to the `signOutTargetUrl`, it's better to merge the new data, preserving existing parameters and avoiding conflicts.

Using `sessionStorage` works too—by reading and clearing the flag immediately, there's no need for a timestamp. However, if the flag isn't cleared correctly, auto functionality might fail on the first login.

Storing a timestamp adds complexity, while query parameters are simpler and visible in the URL. Adjusting query param names if needed is also easier. Clearing the flag after reading ensures smooth operation for future logins without extra overhead.

Both approach works I think query param one is better.